### ISSUE TITLE
ALB-036: BPE tokenizer normalizes whitespace — breaks Python indentation

### ISSUE THREAD

**Issue by noahgift** [paiml/albor#34]
## Summary

The BPE tokenizer trained by `apr tokenize apply` uses `str::split_whitespace()` as the pre-tokenizer, which drops all whitespace (newlines, indentation, blank lines) before BPE merges. This means the vocabulary has no whitespace tokens, and decoded output loses all indentation structure.

## Impact

For Python code models, indentation IS semantics. A tokenizer that normalizes whitespace cannot produce valid Python output. This is acceptable for the 50M pipeline validation but blocks the 350M production model.

## Evidence

```python
encode("def hello():\n    return 42") -> [276, 1154, 924]  # 3 tokens  
decode([276, 1154, 924]) -> "def hello(): return 42"        # indentation lost
```

Vocab has 0 tokens starting with whitespace characters.

## Proposed Fix

Add a `--pre-tokenizer` flag to `apr tokenize apply` with options:
- `whitespace` (current default): `split_whitespace()` — normalizes whitespace
- `byte-level`: ByteLevel pre-tokenizer (like GPT-2, CodeLlama) — preserves all bytes
- `char-level`: Character-level, no pre-tokenization — preserves everything

For code models, `byte-level` is the standard choice (used by GPT-2, CodeGen, StarCoder, CodeLlama).

## Workaround

Train a ByteLevel BPE tokenizer using the Python `tokenizers` library directly:
```python
from tokenizers import Tokenizer, models, trainers, pre_tokenizers
tokenizer = Tokenizer(models.BPE())
tokenizer.pre_tokenizer = pre_tokenizers.ByteLevel(add_prefix_space=False)
trainer = trainers.BpeTrainer(vocab_size=32768, special_tokens=[...])
tokenizer.train(["corpus.txt"], trainer)
tokenizer.save("tokenizer.json")
```

## Components

aprender (text/bpe module), apr CLI
